### PR TITLE
[docs] CToSwift: Add section on methods

### PR DESCRIPTION
**Pull request by jrose-apple**
There's a lot here, so I might have missed something, but I think this document is complete!

### CODE REVIEW

**Comment by jrose-apple** [comment]
@swift-ci Please smoke test

File: docs/CToSwiftNameTranslation.md [hunk 1]
```diff
@@ -212,6 +212,244 @@ Additionally, typedefs for `void *` or `const void *` that are themselves annota
...
+
+    _As an exception, no-argument factory methods cannot have any additional text *after* the matching portion._
+
+3. Accessor methods are associated with a property declaration (`@property`).
+
+4. Subscript methods have one of Objective-C's special subscript selectors (`objectAtIndexedSubscript:`, `objectForKeyedSubscript:`, `setObject:atIndexedSubscript:`, or `setObject:forKeyedSubscript:`).
+
+5. Other methods are imported as plain methods.
+
+If a method overrides a superclass or adopted protocol method, the Swift name of the overridden method will be used for consistency. If there's more than one such name, one is chosen arbitrarily.
```
Comment (varungandhi-apple) [inline docs/CToSwiftNameTranslation.md]: Could you give a small example of how the case of "more than one such name" would arise? Can a single Obj-C method override multiple methods at once?

Comment (jrose-apple) [reply docs/CToSwiftNameTranslation.md]: I guess I can put an example in. It's mostly just `NS_SWIFT_NAME` on one protocol requirement and not on another requirement in a different protocol with the same selector. It's such a terrible situation that I don't want to spend much time on it. :-(

Comment (varungandhi-apple) [reply docs/CToSwiftNameTranslation.md]: I just asked for my own curiosity. I didn't realize you could override multiple methods at once. You don't need to put it in if you think it doesn't warrant discussion.

Comment (jrose-apple) [reply docs/CToSwiftNameTranslation.md]: "override" is a fuzzy term here; it's being used for satisfying protocol requirements too. Maybe that's what should be clarified.

Comment (jrose-apple) [reply docs/CToSwiftNameTranslation.md]: (Clang uses this terminology, but normal Objective-C programmers don't.)

**Comment by jrose-apple** [comment]
@swift-ci Please smoke test

**Comment by jrose-apple** [comment]
@swift-ci Please smoke test macOS

**Comment by jrose-apple** [comment]
@swift-ci Please smoke test macOS

**Comment by jrose-apple** [comment]
@swift-ci Please smoke test macOS

**Pull request merged**